en/docs/deploy-and-publish/deploy-on-gateway/choreo-connect/security/transport-security.md

```diff
@@ -1,26 +1,133 @@
+        - <PATH>/ca-certificates.crt:/etc/ssl/certs/ca-certificates.crt
+    ```
+
+    If you need to change the location of the volume mount, the `trustedCertPath` value in config.toml should also be changed.
+
+1. Convert the certificate into pem format
+ 
+    ```tab="Format"
+        openssl x509 -in <INPUT CERTIFICATE> -out <OUT_PUT_FILE_NAME>.pem
+    ```
+
+    ```tab="Example"
+    openssl x509 -in backend.cert -out backend.pem
+    ```
+
+2. Open the converted backend.pem file and copy the certificate content. (along with `-----BEGIN CERTIFICATE-----` and `-----END CERTIFICATE-----`)
+
+3. Open the default certificate (the mounted certificate) and append the new certificate content and save the file.
+4. Restart the Router container.
+    `docker restart <router_container_name>`
+
+
+## Adding Certificates to Specific Cluters
+
+The above section explains how a certificate can be added to the Choreo Connect router's global trusted certificates. But, if the certificate should be adde only to a specific API backend, it could be done as follows.
+
+#### via API Manager
+
+   Please follow [Adding a Certificate for an Endpoint]({{base_path}}/design/endpoints/certificates/#adding-a-certificate-for-an-endpoint)
+
+<!-- #### via API CTL project
+
+    Please follow -->
+## Securing Communnication between the Components
+
+Communication between internal components of Choreo Connect (adapter, enforcer and router) are secured via mutual ssl.
+
+Each component has it's private key and truststore and it's configured in against each component tag `config.toml` file as below.
+
+```toml
+    [adapter.keystore] 
+    certPath = "/home/wso2/security/keystore/mg.pem"
+    keyPath = "/home/wso2/security/keystore/mg.key"
+
+    [adapter.truststore]
+    location = "/home/wso2/security/truststore"
+```
+
+## Adding a Certificate to Adapter/ Enforcer Truststore
+
+The trusted certificate location is configured as a volume mount for Adapter and Enforcer components in the docker-compose.yaml file as below.
+
+``` tab="docker-compose.yaml"
+volumes:
+        - ../resources/adapter/security:/home/wso2/security
+```
+
+To add a new certificate Choreo Connect component, it should be done by adding the pem formated certificate to the truststore location of that particular component.
+
+For an example, if a new certificate needs to be added to the adapter component,
+
+1. The certificate should be converted to PEM formate if it is not.
+2. Copy the PEM certificate into `<CHOREO-CONNECT_HOME>/resources/adapter/security/truststore` directory.
+3. Restart the adapter container.
+
+## Changing the Private Certificate of a Component
+If it is required to change the private certificate of a component, follow the steps below.
+
+1. Copy the private key and certificate (in PEM format) into `<CHOREO-CONNECT_HOME>/resources/<COMPONENT>/security/keystore` directory and `<CHOREO-CONNECT_HOME>/resources/<COMPONENT>/security/truststore` directory.
```

Comment:
shall we simply ask to replace the mg.key and mg.pem files within each directory? That would be easier to grab I guess.

Then after let's put a note saying that if the file name is different change the 

- adapter's case, it is config.toml 
- route's case there are two env variables expecting (enforcer_cert and adapter_cert) for adapter and enforcer, for its own private-public key pair there is another two.

- enforcer's case also env variables.